src_features/provideNFTInformation/cmd_provideNFTInfo.c

```diff
@@ -159,20 +159,10 @@ void handleProvideNFTInformation(uint8_t p1,
 
     uint8_t algorithmId = workBuffer[offset];
     PRINTF("Algorithm: %d\n", algorithmId);
-    cx_curve_t curve;
-    verificationAlgo *verificationFn;
-    cx_md_t hashId;
-
-    switch (algorithmId) {
-        case ALGORITHM_ID_1:
-            curve = CX_CURVE_256K1;
-            verificationFn = (verificationAlgo *) cx_ecdsa_verify;
-            hashId = CX_SHA256;
-            break;
-        default:
-            PRINTF("Incorrect algorithmId %d\n", algorithmId);
-            THROW(0x6a80);
-            break;
+
+    if (algorithmId != ALGORITHM_ID_1) {
+        PRINTF("Incorrect algorithmId %d\n", algorithmId);
+        THROW(0x6a80);
```

Comment:
do we have a defined apdu code for this ?

Reply:
Yes, we have this one:
`#define APDU_RESPONSE_INVALID_DATA            0x6a80`

I have just kept the same logic as most of the code in this file... But Ok, I will change the hardcoded values by the define